pkg/binding/bindingpolicy-resolver.go

```diff
@@ -185,13 +184,14 @@ func (resolver *bindingPolicyResolver) NoteBindingPolicy(bindingpolicy *v1alpha1
 	resolver.createResolution(bindingpolicy)
 }
 
-// NoteObject updates the maintained bindingpolicy's objects resolution for the
-// given bindingpolicy key. If the object is being deleted, it is removed from
-// the resolution if exists.
+// AddObjectIdentifier adds an object's identifier to the resolution for
+// the given bindingpolicy key. If the object is already in the resolution,
+// it is not added again.
 //
 // The returned bool indicates whether the bindingpolicy resolution was changed.
-func (resolver *bindingPolicyResolver) NoteObject(bindingPolicyKey string,
-	obj runtime.Object) (bool, error) {
+// If no resolution is associated with the given key, an error is returned.
+func (resolver *bindingPolicyResolver) AddObjectIdentifier(bindingPolicyKey string,
```

Comment:
FYI: "Add" is not quite the right word here, since we are explicitly contemplating the possibility that the ObjectIdentifier is already included. I would use something like "Ensure".